tests/src/registry.spec.ts

```diff
@@ -0,0 +1,143 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { Page } from 'playwright';
+import { PodmanDesktopRunner } from './runner/podman-desktop-runner';
+import { afterAll, beforeAll, test, describe, beforeEach } from 'vitest';
+import { expect as playExpect } from '@playwright/test';
+import type { RunnerTestContext } from './testContext/runner-test-context';
+import { WelcomePage } from './model/pages/welcome-page';
+import { SettingsBar } from './model/pages/settings-bar';
+import { RegistriesPage } from './model/pages/registries-page';
+import { NavigationBar } from './model/workbench/navigation';
+
+let pdRunner: PodmanDesktopRunner;
+let page: Page;
+let navBar: NavigationBar;
+let registryUrl: string;
+let registryUsername: string;
+let registryPswdSecret: string;
+let registryName: string;
+
+beforeAll(async () => {
+  pdRunner = new PodmanDesktopRunner();
+  page = await pdRunner.start();
+  pdRunner.setVideoAndTraceName('registry-e2e');
+
+  registryUrl = process.env.REGISTRY_URL ? process.env.REGISTRY_URL : process.env.CI ? 'ghcr.io' : '';
+  registryUsername = process.env.REGISTRY_USERNAME
```

Comment:
Extract this logic into what you got in setup-files. So it is unified across both tests.